src/app/file-manager/services/files.service.ts

```diff
@@ -95,47 +97,42 @@ export class FilesService {
      */
     private createFileNode(treeLocal: TreeNode[], item: any): TreeNode[] {
         const node: TreeNode = {};
-        const fileData: FileData = {
-            id: item.id,
-            selfLink: item.selfLink,
-            bucket: item.bucket,
-            created: item.timeCreated,
-            updated: item.updated,
-            path: item.name.split('/')[item.name.split('/').length - 1],
-            size: item.size + ' MB',
-            type: (<string>item.name).endsWith('/') ? 'Folder' : 'File',
-            leaf: true
-        };
+        const fileData: FileData = this.getFileData(item, item.name.split('/')[item.name.split('/').length - 1]);
         node.data = fileData;
         treeLocal.push(node);
         return treeLocal;
     }
 
     private createNode(item: any, path: string): TreeNode {
         const node: TreeNode = {};
+        const fileData: FileData = this.getFileData(item, path);
+
+        node.data = fileData ;
+        return node;
+    }
+
+    private getFileData(item: any, path: string) {
         const fileData: FileData = {
             id: item.id,
             selfLink: item.selfLink,
             bucket: item.bucket,
             created: item.timeCreated,
             updated: item.updated,
             path: path,
-            size: item.size + ' MB',
+            size: this.filterSize.transform(item.size),
```

Comment:
This works but I thought the angular way to do this was to use a custom filter, and use a filter in the UI, not modify the data.

Reply:
Given the way we show file data using a primeng's p-treeTable : 

<p-treeTable [value]="files" selectionMode="checkbox" [(selection)]="selectedFiles" styleClass="tree-table" (onNodeSelect)="nodeSelect($event)"
  (onNodeUnselect)="nodeUnselect($event)">
  *<p-column *ngFor="let col of cols " [field]="col.field" [header]="col.header"></p-column>*
</p-treeTable>
we should explicitly create one p-column for each field to be able to use pipes or other UI formatting to the size field. Currently, we can modify which columns are shown just changing cols array. 

We found it is more appropriate to modify data once and use it N times instead of transforming data on each place where it is shown.

If you still prefer to have pipes / formatting at column level instead of data array, please let me know and I will change it asap.

Reply:
OK, I understand. I'm more used to seeing code where the data exported to the UI isn't formatted or filtered and the formatting is done in the HTML code. It's OK to leave it this way for now.